x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/Foldables.java

```diff
@@ -46,11 +49,18 @@ public static double doubleValueOf(Expression e) {
     }
 
     public static <T> List<T> valuesOf(List<Expression> list, DataType to) {
-        List<T> l = new ArrayList<>(list.size());
-        for (Expression e : list) {
-             l.add(valueOf(e, to));
+        return foldAndAdd(list, to, new ArrayList<>(list.size()));
+    }
+
+    public static <T> Set<T> valuesOfNoDuplicates(List<Expression> list, DataType to) {
+        return foldAndAdd(list, to, new LinkedHashSet<>(list.size()));
+    }
+
+    private static <T, C extends Collection<T>> C foldAndAdd(Collection<Expression> expressions, DataType to, C values) {
```

Comment:
Please rename it to `foldTo` - I was thinking the values are folded and added together based on the current name.